ctci/chapter2/2.2.py

```diff
@@ -0,0 +1,57 @@
+#2.2
+#Return Kth to Last: Implement an algorithm 
+#to find the kth to last element of a singly linked list.
+
+#assuming no cycles
+
+from random import randrange
+
+class Node:
+    def __init__(self, dataval=None):
+        self.dataval = dataval
+        self.nextval = None
+
+class SLinkedList:
+    def __init__(self):
+        self.headval = None
+
+
+def kthLast (h: SLinkedList, k: int):
+
+    v = None
+    
+    def iterator (n: Node):
+        nonlocal v
+        if (n == None):
+            return 0
+        else:
+            count = iterator(n.nextval)
```

Comment:
This has to be the most confusing way to solve this problem.
Either use two pointers `A,B`, move `B` forward k units and then move both until `B` has reached the end.
Or
Get the length of the linked list like a non degenerate using regular recursion or a for loop

Reply:
It may also be beneficial for me + mike, if you wrote a little doc string explaining what your doing (for more complex problems, think its fine to not have one for these problems)